toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -57,10 +60,179 @@ func handleKernelCommandLine(extraCommandLine string, imageChroot *safechroot.Ch
 	newGrub2ConfigFile := grub2ConfigFile[:insertIndex] + extraCommandLine + " " + grub2ConfigFile[insertIndex:]
 
 	// Update grub.cfg file.
-	err = os.WriteFile(grub2ConfigFilePath, []byte(newGrub2ConfigFile), 0)
+	err = writeGrub2ConfigFile(newGrub2ConfigFile, imageChroot)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Updates the kernel command-line args with the new SELinux mode.
+//
+// See, installutils.setGrubCfgSELinux()
+func updateSELinuxCommandLine(selinuxMode imagecustomizerapi.SELinux, imageChroot *safechroot.Chroot) error {
+	logger.Log.Infof("Updating SELinux kernel command-line args")
+
+	grub2Config, err := readGrub2ConfigFile(imageChroot)
+	if err != nil {
+		return err
+	}
+
+	// Try to find any existing SELinux args.
+	selinuxArgsMatch := linuxCommandLineSELinuxRegex.FindStringSubmatchIndex(grub2Config)
+
+	start := 0
+	end := 0
+
+	if selinuxArgsMatch == nil {
+		// SELinux args are not in the grub.cfg file.
+		if selinuxMode == imagecustomizerapi.SELinuxDisabled {
+			// Nothing to do.
+			return nil
+		}
+
+		insertLocationMatch := linuxCommandLineInsertRegex.FindStringIndex(grub2Config)
+		if insertLocationMatch == nil {
+			return fmt.Errorf("failed to find location to insert SELinux kernel command-line args")
+		}
+
+		// Use the end of the regex match as the point to do the insert at.
+		insertPoint := insertLocationMatch[1]
+		start = insertPoint
```

Comment:
With the comment and the `insertLocationMatch` variable name, do we even need the separate `insertPoint`?
```suggestion
		start = insertLocationMatch[1]
```